src/blame.c

```diff
@@ -451,7 +455,12 @@ blame_request(struct view *view, enum request request, struct line *line)
 			if (diff->pipe)
 				string_copy_rev(diff->ref, NULL_ID);
 		} else {
-			open_diff_view(view, flags);
+			if (filename_from_blame) {
+				free(filename_from_blame);
```

Comment:
Somehow, I don't remember the pattern of "free the memory all but one time" in Kerninghan and Ritchie. 😢

Perhaps I should write to them and they'll put it into the next edition.

Reply:
OK, I think I know how to fix this: simply access the global `blame_view` from `diff.c` to get this info. I may or may not get to it before I hear something back about whether the project is interested in this functionality.